pylint/checkers/strings.py

```diff
@@ -395,10 +398,54 @@ def visit_binop(self, node: nodes.BinOp) -> None:
                             args=(arg_type.pytype(), format_type),
                         )
 
-    @only_required_for_messages("f-string-without-interpolation")
+    @only_required_for_messages(
+        "f-string-without-interpolation", "bad-format-character"
+    )
     def visit_joinedstr(self, node: nodes.JoinedStr) -> None:
+        node_string = node.as_string()[2 : len(node.as_string()) - 1]
+        format_map = {}
+        try:
+            format_map = utils.parse_all_fields_formatting(node_string)
+        except utils.UnsupportedFormatCharacter as exc:
+            formatted = node_string[exc.index]
+            self.add_message(
+                "bad-format-character",
+                node=node,
+                args=(formatted, ord(formatted), exc.index),
+            )
+            return
+        except utils.IncompleteFormatString:
+            self.add_message("bad-format-string", node=node)
```

Comment:
We don't have the coverage report yet, but the fact that `bad-format-string` was missing in the `@only_required_for_messages` decorator and we don't have a test failure for the test run with `--minimal-messages-config` means that this line is not yet covered by tests.